### PR TITLE
Add pod and service labels support

### PR DESCRIPTION
**Pull request by halkeye**
#### What this PR does / why we need it:
Add pod and service labels.

#### Special notes for your reviewer:
@bertinwong-saucelabs asked me to do it cause I had a bit of spare time.
He was planning on installing the push gateway but needed some extra
labels for the k8s cluster he was working on

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [X] Variables are documented in the README.md

### CODE REVIEW

File: stable/prometheus-pushgateway/templates/deployment.yaml [hunk 1]
```diff
@@ -1,12 +1,12 @@
+{{- $labelChart := include "prometheus-pushgateway.chart" $ -}}
+{{- $labelApp := include "prometheus-pushgateway.name" $ -}}
```
Comment (gianrubio) [inline stable/prometheus-pushgateway/templates/deployment.yaml]: please move this to the _helper.tmpl, ex https://github.com/helm/charts/blob/master/stable/prometheus-node-exporter/templates/_helpers.tpl#L27-L36

**Comment by halkeye** [comment]
@gianrubio I did it, but in a slightly different way. I realized it afterwards. Do you want me to try and go back? Should I have 1 config for extra labels? or the same labels for all 3 types?

**Comment by gianrubio** [comment]
Can you tell me if there any difference in the labels when you run helm upgrade from the current version to this one?

**Comment by halkeye** [comment]
I didn't think to actually check the differences.

```
  add-labels-and-annotations ?  …/charts/stable/prometheus-pushgateway  diff -u original.yaml <(helm template .)
--- original.yaml       2018-10-29 12:57:37.745053971 -0700
+++ /proc/self/fd/11    2018-10-29 12:59:06.003485841 -0700
@@ -6,9 +6,10 @@
   name: RELEASE-NAME-prometheus-pushgateway
   labels:
     app: prometheus-pushgateway
-    chart: prometheus-pushgateway-0.1.2
-    release: "RELEASE-NAME"
-    heritage: "Tiller"
+    chart: prometheus-pushgateway-0.1.3
+    heritage: Tiller
+    release: RELEASE-NAME
+
 ---
 # Source: prometheus-pushgateway/templates/service.yaml
 apiVersion: v1
@@ -17,9 +18,10 @@
   name: RELEASE-NAME-prometheus-pushgateway
   labels:
     app: prometheus-pushgateway
-    chart: prometheus-pushgateway-0.1.2
-    release: RELEASE-NAME
+    chart: prometheus-pushgateway-0.1.3
     heritage: Tiller
+    release: RELEASE-NAME
+
 spec:
   type: ClusterIP
   ports:
@@ -39,9 +41,10 @@
   name: RELEASE-NAME-prometheus-pushgateway
   labels:
     app: prometheus-pushgateway
-    chart: prometheus-pushgateway-0.1.2
-    release: RELEASE-NAME
+    chart: prometheus-pushgateway-0.1.3
     heritage: Tiller
+    release: RELEASE-NAME
+
 spec:
   replicas: 1
   selector:
@@ -81,6 +84,7 @@
           resources:
             {}

+
 ---
 # Source: prometheus-pushgateway/templates/ingress.yaml
```

Looks like heritage and release got switched. I might be able to swap them, but I suspect dict is ordering them. and it doesn't look like heritage is quoted anymore.

**Comment by gianrubio** [comment]
order doesn't matter for labels so I'm happy to merge it.

/ok-to-test

**Comment by halkeye** [comment]
rebased (merge got huge and messy)

**Comment by gianrubio** [comment]
/ok-to-test
/lgtm
thanks @halkeye

**Pull request merged**